### PR TITLE
feature-2-login

### LINKED ISSUE #2: [Feature 2] LogIn
LogIn a user.
 - Enter an email
 - Enter a password
 - Button submit
 - Show errors such as password doesn't match or no email found.
 - Show Success if LogIn attempt was successful

### PR DESCRIPTION
**Pull request by RyanSafa**
Implemented the user login use case

- Users can enter an email and a password
- Users are authenticated if there is a matching pair of email and passwords in the database
- If there is not a match, then the users are sent an error message

Implemented clean architecture to create this use case. Also used the singleton design pattern to represent the currently logged in user.

### CODE REVIEW

File: src/main/java/intefaceAdapters/userLogin/FindUser.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        BufferedReader reader = new BufferedReader(new FileReader(csvFile));
+        reader.readLine(); // skip header
+
+        String row;
+        while ((row = reader.readLine()) != null) {
+            String[] col = row.split(",");
+            String email = String.valueOf(col[headers.get("email")]);
+            String password = String.valueOf(col[headers.get("password")]);
+            accounts.put(email,password);
+        }
```
Comment (manav393) [inline src/main/java/intefaceAdapters/userLogin/FindUser.java:43]: This piece of code is too long to be in a constructor for a class. I recommend keeping the constructor clean and as simple as possible by extracting the functionality to a helper method.

File: src/main/java/useCases/userLogin/UserLoginInteractor.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+
+    @Override
+    public void logInUser(UserLoginRequestModel requestModel) {
+        if (userLoginDsGateway.validLogin(requestModel.getEmail(), requestModel.getPassword())) {
+            UserLoginResponseModel userLoginResponseModel = new UserLoginResponseModel("Welcome back to Ryde!", requestModel.getEmail());
+            LoggedInUserSingleton.init(requestModel.getEmail());
+            userLoginPresenter.loginSuccess(userLoginResponseModel);
+        } else {
```
Comment (manav393) [inline src/main/java/useCases/userLogin/UserLoginInteractor.java:41]: This piece is kind of hard to read. I would recommend extracting the email and password as separate variables from the request model and then inputting them as parameters.

**Review by manav393 — COMMENTED** [review]
Before merging this into the main branch, I think you should finish adding the javadocs. As for the unit tests, we can add those toward the end of our project.

**Comment by RyanSafa** [comment]
Done with all your requests. Thanks for the code review.

**Review by manav393 — APPROVED** [review]
Looks good ryan!

**Review by jyotiraditya09 — APPROVED** [review]
Good work!

**Pull request merged**